pkg/arvo/sys/vane/ames.hoon

```diff
@@ -4034,7 +4051,7 @@
             %-  (fi-trace fin.veb |.("done {(spud fi-full-path)}"))
             %-  ~(rep in listeners.keen)
             |=  [=^duct =_fine]
-            fine(event-core (emit:fine duct %give gift))
```

Comment:
@yosoyubik Since you removed this and Joe later had to readd it, it seems worth to explicitly explain the pattern:

Since we're accumulating results in new versions of `fine`, we need to call the `+emit` that is available through the accumulated `fine`, not the one that we have in scope in the initial version. Otherwise only the last gift will actually be given, since we're overwriting `event-core`'s state every time we emit.

Reply:
Thanks! Yeah, the removal was a spurious one, and it shouldn't have happened—the pattern I had used before (e.g. +fi-error, +fi-clean-up) was making the core in the accumulator the event core, not fine, since there was no emit arm there. I added one now https://github.com/urbit/urbit/pull/6406/commits/5c838a42a795a83f651e703d0d312c08ef4103f0 so hopefully that's more clear.